app/validators/password_integrity_validator.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+require 'zxcvbn'
+class PasswordIntegrityValidator < ActiveModel::EachValidator
+  def validate_each(record, attr_name, value)
+    return if record.admin?
+    return if value.nil? and record.encrypted_password.present?
+
+    password_test = Zxcvbn.test(value, [record.email, record.phone_number, record.name])
+
+    # Scoring of the test is a range of 0 - 4
+    #   0: too guessable: risky password. (guesses < 10^3)
+    #   1: very guessable: protection from throttled online attacks. (guesses < 10^6)
+    #   2: somewhat guessable: protection from unthrottled online attacks. (guesses < 10^8)
+    #   3: safely unguessable: moderate protection from offline slow-hash scenario. (guesses < 10^10)
+    #   4: very unguessable: strong protection from offline slow-hash scenario. (guesses >= 10^10)
+    # (from https://github.com/dropbox/zxcvbn#usage)
+
+    # NOTE: Add note of feedback from `password_test.feedback`?
+    puts JSON.pretty_generate(password_test.feedback.suggestions) if Rails.env.development?
+    record.errors.add(attr_name, I18n.t("errors.attributes.password.insecure", feedback: password_test.feedback.suggestions)) if password_test.score <= 2
```

Comment:
when we tried this it just said we need to "increase the complexity" -- seems like we would need some advice about what this means -- longer password, more types of characters, whatever. otherwise the error may frustrate people who don't know how it needs to change